cassdegrees/ui/views.py

```diff
@@ -442,24 +447,42 @@ def manage_courses(request):
                     render_properties['edit_course_info'] = current_course_info
 
                 else:
-                    render_properties['is_error'] = True
+                    render_properties['msg_type'] = 'msg-error'
                     render_properties['hide_form'] = True
                     render_properties['msg'] = "Please select a course to edit!"
 
             elif action == 'Delete':
-                ids_to_delete = post_data.getlist('id')
                 rest_api = None
+                ids_to_delete = post_data.getlist('id')
+                used_subplans = []
+                courses_in_subplans = list(CoursesInSubplanModel.objects.all().values())
                 for id_to_delete in ids_to_delete:
-                    rest_api = requests.delete(model_api_url + id_to_delete + '/')
-
-                if rest_api is None:
-                    render_properties['is_error'] = True
+                    if perform_function == 'confirm deletion':  # if user has clicked 'yes' on the confirmation page
+                        rest_api = requests.delete(model_api_url + id_to_delete + '/')
+                    else:
+                        if not courses_in_subplans:  # delete immediately if no subplans use the course
+                            rest_api = requests.delete(model_api_url + id_to_delete + '/')
+                        else:
+                            for course in courses_in_subplans:
+                                # if course being deleted is in current subplan
+                                if int(id_to_delete) == int(course['courseId_id']):
+                                    # TODO: improve this using django queries
+                                    used_subplans.extend(
+                                        [subplan['name'] + '(' + subplan['code'] + ')' for subplan in subplans if
+                                         course['subplanId_id'] == subplan['id']])
+                if used_subplans:  # if there any subplans that use the course
+                    render_properties['is_confirm'] = True
+                    render_properties['msg_type'] = 'msg-warn'
+                    render_properties['msg'] = 'The Sub-Plans ' + ', '.join(used_subplans) + ' use this course. Do you want to continue?'
```

Comment:
```suggestion
                    render_properties['msg'] = 'The Sub-Plan(s) ' + ', '.join(used_subplans) + ' use this course. Do you want to continue?'
```

Minor styling change

Reply:
It should also be noted that this approach will no longer work once Aiden's PR (#76) is accepted, as he is deleting the CoursesInSubplan table. Refer to that PR for a new solution

Reply:
will do.